src/Common/Commands.Common/AzurePSCmdlet.cs

```diff
@@ -137,13 +137,45 @@ protected string PSVersion
 
         protected abstract string DataCollectionWarning { get; }
 
+        private SessionState _asJobSessionState;
+
+        public SessionState AsJobSessionState
```

Comment:
I wonder if we shouldn't Hide SessionState, and throw if it is used, thoughts?

Reply:
In fact, I'm wondering if my previous advice about not hiding was good.  If we can get the propertes using ```base.```, then hiding them would make the interface simpler. 

Otherwise, we should just rething the name of CopiedParameterDictionary (maybe AsJobParameterDictionary)

Reply:
I think it makes sense to hide it here - updated accordingly.